nsv13/code/modules/munitions/ship_weapons/ballistic_weapons/revision2/ammo_rack.dm

```diff
@@ -0,0 +1,375 @@
+	if(!jammed && default_deconstruction_crowbar(I))
+		return
+	// Smacking
+	if(user.a_intent == INTENT_HARM)
+		return
+	if(busy)
+		to_chat(user, "<span class='warning'>Someone's already working on [src]!</span>")
+		return TRUE
+	if(panel_open && istype(I, /obj/item/reagent_containers))
+		if(jammed)
+			to_chat(user, "<span class='warning'>You can't lubricate a jammed machine!</span>")
+			return TRUE
+		if(durability >= 100)
+			to_chat(user, "<span class='warning'>[src] doesn't need any oil right now!</span>")
+			return TRUE
+		if(!I.reagents.has_reagent(/datum/reagent/oil))
+			to_chat(user, "<span class='warning'>You need oil to lubricate this!</span>")
+			return TRUE
+		// get how much oil we have
+		var/oil_amount = min(I.reagents.get_reagent_amount(/datum/reagent/oil), max_durability/repair_multiplier)
+		var/oil_needed = CLAMP(ROUND_UP((max_durability-durability)/repair_multiplier), 1, oil_amount)
+		oil_amount = min(oil_amount, oil_needed)
+		user.visible_message("<span class='notice'>[user] begins lubricating [src]...</span>", \
+					"<span class='notice'>You start lubricating the inner workings of [src]...</span>")
+		busy = TRUE
+		if(!do_after(user, 5 SECONDS, target=src))
+			busy = FALSE
+			to_chat(user, "<span class='warning'>You were interrupted!</span>")
+			return TRUE
+		if(!I.reagents.has_reagent(/datum/reagent/oil, oil_amount)) //things can change, check again.
+			to_chat(user, "<span class='warning'>You don't have enough oil left to lubricate [src]!</span>")
+			busy = FALSE
+			return TRUE
+		user.visible_message("<span class='notice'>[user] lubricates [src].</span>", \
+					"<span class='notice'>You lubricate the inner workings of [src].</span>")
+		durability = min(durability + (oil_amount * repair_multiplier), max_durability)
+		I.reagents.remove_reagent(/datum/reagent/oil, oil_amount)
+		busy = FALSE
+		return TRUE
+	if(jammed && I.tool_behaviour == TOOL_CROWBAR)
+		if(panel_open)
+			to_chat(user, "<span class='notice'>You need to close the panel to get at the jammed machinery.</span>")
+			return TRUE
+		busy = TRUE
+		user.visible_message("<span class='notice'>[user] begins clearing the jam in [src].</span>", \
+					"<span class='notice'>You being clearing the jam in [src].</span>")
+		if(!do_after(user, 10 SECONDS, target=src))
+			busy = FALSE
+			to_chat(user, "<span class='warning'>You were interrupted!</span>")
+			return TRUE
+		user.visible_message("<span class='notice'>[user] clears the jam in [src].</span>", \
+					"<span class='notice'>You clear the jam in [src].</span>")
+		playsound(src, 'nsv13/sound/effects/ship/mac_load_unjam.ogg', 100, 1)
+		jammed = FALSE
+		durability += rand(0,5) //give the poor fools a few more uses if they're lucky
+		busy = FALSE
+		return TRUE
+	if(istype(I, /obj/item/ship_weapon/ammunition) || istype(I, /obj/item/powder_bag))
+		to_chat(user, "<span class='notice'>You start to load [src] with [I].</span>")
+		if(!do_after(user, 1 SECONDS , target = src))
```

Comment:
I'm not sure if 1 second is fast enough, considering the current Bumped() interaction makes loading by just throwing faster.